packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -167,11 +188,32 @@ class AndroidValidator extends DoctorValidator {
         messages.add(ValidationMessage.error('Android license status unknown.'));
         return ValidationResult(ValidationType.partial, messages, statusInfo: sdkVersionText);
     }
-
-    // Success.
     return ValidationResult(ValidationType.installed, messages, statusInfo: sdkVersionText);
   }
 
+  bool _checkJavaVersionNoOutput() {
+    final String javaBinary = AndroidSdk.findJavaBinary();
+    if (javaBinary == null) {
+      return false;
+    }
+    if (!processManager.canRun(javaBinary)) {
+      return false;
+    }
+    String javaVersion;
+    try {
+      final ProcessResult result = processManager.runSync(<String>[javaBinary, '-version']);
+      if (result.exitCode == 0) {
+        final List<String> versionLines = result.stderr.split('\n');
+        javaVersion = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
+      }
+    } catch (_) { /* ignore */ }
```

Comment:
You should at least `printTrace()` this error

Reply:
Done. Did the same for the base Android validator.